EngineLayer/ClassicSearch/ClassicSearchEngine.cs

```diff
@@ -100,12 +103,29 @@ protected override MetaMorpheusEngineResults RunSpecific()
                         // digest each protein into peptides and search for each peptide in all spectra within precursor mass tolerance
                         foreach (PeptideWithSetModifications peptide in Proteins[i].Digest(CommonParameters.DigestionParams, FixedModifications, VariableModifications, SilacLabels, TurnoverLabels))
                         {
-                            PeptideWithSetModifications reversedOnTheFlyDecoy = null;
-
-                            if (SpectralLibrary != null)
+                            PeptideWithSetModifications generatedOnTheFlyDecoy = null;
+                            // Do rev check similarity, do scrambled, check sim, do mirrored
+                            if (SpectralLibrary != null || DecoyOnTheFly == true)
```

Comment:
This change causes decoys to be generated differently when using a spectral library, causing the TestReverseDecoyGenerationDuringSearch unit test to fail.